src/__internal__/checkable-input/checkable-input.component.tsx

```diff
@@ -62,119 +62,123 @@ export interface CheckableInputProps extends CommonCheckableInputProps {
-    fieldHelpInline,
-    fieldHelpId,
-    id,
-    info,
-    label,
-    labelAlign,
-    labelHelp,
-    labelHelpIcon: "info" as const,
-    labelId,
-    labelInline,
-    labelSpacing,
-    name: id,
-    reverse,
-    warning,
-    validationIconId,
-    // We don't want an asterisk on each radio control, only the legend
-    // However, we still want the input element to receive the required prop
-    isRequired: isRadio ? undefined : required,
-    useValidationIcon: validationOnLabel,
-  };
+    const formFieldProps = {
+      disabled,
+      error,
+      fieldHelp,
+      fieldHelpInline,
+      fieldHelpId,
+      id,
+      info,
+      label,
+      labelAlign,
+      labelHelp,
+      labelHelpIcon: "info" as const,
+      labelId,
+      labelInline,
+      labelSpacing,
+      name: id,
+      reverse,
+      warning,
+      validationIconId,
+      // We don't want an asterisk on each radio control, only the legend
+      // However, we still want the input element to receive the required prop
+      isRequired: isRadio ? undefined : required,
+      useValidationIcon: validationOnLabel,
+    };
 
-  const inputProps = {
-    "aria-describedby": ariaDescribedBy,
-    "aria-labelledby": externalAriaLabelledBy || ariaLabelledBy,
-    "aria-invalid": !!error,
-    autoFocus,
-    checked,
-    disabled,
-    id,
-    inputRef,
-    type,
-    value,
-    name,
-    onBlur,
-    onChange,
-    onFocus,
-    required,
-    ...props,
-  };
+    const inputProps = {
+      "aria-describedby": ariaDescribedBy,
+      "aria-labelledby": externalAriaLabelledBy || ariaLabelledBy,
+      "aria-invalid": !!error,
+      autoFocus,
+      checked,
+      disabled,
+      id,
+      type,
+      value,
+      name,
+      onBlur,
+      onChange,
+      onFocus,
+      required,
+      ref,
+      ...props,
+    };
 
-  return (
-    <StyledCheckableInputWrapper
-      disabled={disabled}
-      fieldHelpInline={fieldHelpInline}
-      inputWidth={inputWidth}
-      labelWidth={labelWidth}
-      labelInline={labelInline}
-      reverse={reverse}
-    >
-      <InputBehaviour>
-        <FormField {...formFieldProps}>
-          <StyledCheckableInput>
-            <HiddenCheckableInput {...inputProps} />
-            {children}
-          </StyledCheckableInput>
-        </FormField>
-      </InputBehaviour>
-    </StyledCheckableInputWrapper>
-  );
-};
+    return (
+      <StyledCheckableInputWrapper
+        disabled={disabled}
+        fieldHelpInline={fieldHelpInline}
+        inputWidth={inputWidth}
+        labelWidth={labelWidth}
+        labelInline={labelInline}
+        reverse={reverse}
+      >
+        <InputBehaviour>
+          <FormField {...formFieldProps}>
+            <StyledCheckableInput>
+              <HiddenCheckableInput {...inputProps} />
+              {children}
+            </StyledCheckableInput>
+          </FormField>
+        </InputBehaviour>
+      </StyledCheckableInputWrapper>
+    );
+  }
+);
```

Comment:
yeah, I noticed sometimes it was causing TS errors, but not others. I'm not sure why it would cause a TS error actually, although it's useful when it does! I meant to go back at the end and make sure I added `displayName` to all the components I'd added `forwardRef` to but I think I forgot some - I'll make sure I fix that!